.github/workflows/run-post-submission-jobs.yaml

```diff
@@ -20,11 +20,50 @@ permissions:
 
 jobs:
 
+  get-all-dates:
+    runs-on: ubuntu-latest
+    outputs:
+      NOWCAST_DATE: ${{ steps.set-dates.outputs.NOWCAST_DATE }}
+      SEQUENCE_AS_OF: ${{ steps.set-dates.outputs.SEQUENCE_AS_OF }}
+      matrix: ${{ steps.set-dates.outputs.MATRIX_DATES }}
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v4
+        with:
+            # only checkout what we need
+            sparse-checkout: |
+              .github/
+
+      - name: Set dates 🕰️
+        id: set-dates
+        # Nowcast_date = round_id of the round that just closed for submissions.
+        # Sequence_as_of = date used to get the sequences for clade assignment.
+        # Matrix_dates = dates used for matrix in create-target-data job (including
+        # the round_id of the 91-day-old round that can now be officially closed).
+        run: |
+          source ${GITHUB_WORKSPACE}/.github/shared/shared-functions.sh
+          nowcast_date=$(get_latest_wednesday "${{ inputs.nowcast_date }}")
+          sequence_as_of=$(date -d "$nowcast_date - 1 day" +%Y-%m-%d)
+          matrix_dates=$(generate_weekly_dates "$nowcast_date" 13)
+          echo "nowcast_date=$nowcast_date" >> $GITHUB_OUTPUT
+          echo "sequence_as_of=$sequence_as_of" >> $GITHUB_OUTPUT
+          echo "matrix_dates=$matrix_dates" >> $GITHUB_OUTPUT
+
+          echo -e "\nDebug values:"
+          echo "Nowcast date: $nowcast_date"
+          echo "Sequence as of: $sequence_as_of"
+          echo "Matrix dates: $matrix_dates"
+
   # Create target data for the round that closed 90 days ago.
   # This step will exit if there is no round with a nowcast
   # date 90 days ago.
   create-target-data:
     runs-on: ubuntu-latest
+    needs: get-all-dates
+    env:
+      SEQUENCE_AS_OF: ${{ needs.get-all-dates.outputs.sequence_as_of }}
+    strategy:
+      matrix: ${{ fromJson(needs.get-all-dates.outputs.matrix) }}
```

Comment:
Matrix is created using the output of the new `generate_weekly_dates()` function